1) Em um concurso de arte com 15 participantes, as notas das avaliacoes dos quadros de cada um deles foram:

10, 6, 9, 8, 9, 8, 7, 7, 5, 10, 9, 8, 5, 3, 6.

No entanto, para a divulgacao dos resultados pelo marketing da galeria, sera necessario transformar esses valores nos conceitos A, B, C e D. O conceito D sera para notas abaixo de 5, conceito C para notas de 5 a 7, conceito B para notas de 7 a 9, e conceito A para notas acima de 9.

Calcule e construa uma tabela de distribuicao de frequencias para melhor visualizacao do desempenho dos concorrentes. 

In [1]:
# Deixe seus calculos aqui:
lista_de_notas = [10, 6, 9, 8, 9, 8, 7, 7, 5, 10, 9, 8, 5, 3, 6]
minimo = min(lista_de_notas)
maximo = max(lista_de_notas)
#print(minimo, maximo)
print("Mínimo:",minimo)
print("Máximo:",maximo)

Mínimo: 3
Máximo: 10


In [2]:
#Calcular a amplitude
amplitude = maximo - minimo
print(amplitude)

7


In [3]:
# se precisassemos calcular o numero de classes
# achar os intervalos = dividir o numero da amplitude pelo numero de classes
intervalo = 7/4
print(intervalo)

#arredonadando o intervalo de 1,75 passa para 2
intervalo = 2

#frequencia = soma de elementos de mesmo valor
frequencia_d = sum(1 for item in lista_de_notas if item < 5)
frequencia_c = sum(1 for item in lista_de_notas if item >= 5 and item < 7)
frequencia_b = sum(1 for item in lista_de_notas if item >= 7 and item < 9)
frequencia_a = sum(1 for item in lista_de_notas if item >= 9)

print("Frequencia d:",frequencia_d)
print("Frequencia c:",frequencia_c)
print("Frequencia b:",frequencia_b)
print("Frequencia b:",frequencia_a)

1.75
Frequencia d: 1
Frequencia c: 4
Frequencia b: 5
Frequencia b: 5


In [4]:
#Calcula frequencia relativa
frequencia_relativa_d = frequencia_d / len(lista_de_notas)
frequencia_relativa_c = frequencia_c / len(lista_de_notas)
frequencia_relativa_b = frequencia_b / len(lista_de_notas)
frequencia_relativa_a = frequencia_a / len(lista_de_notas)
print(f"{(frequencia_relativa_d * 100):0.2f}")
print(f"{(frequencia_relativa_c * 100):0.2f}")
print(f"{(frequencia_relativa_b * 100):0.2f}")
print(f"{(frequencia_relativa_a * 100):0.2f}")

total = frequencia_relativa_d + frequencia_relativa_c + frequencia_relativa_b + frequencia_relativa_a
print(total)

6.67
26.67
33.33
33.33
1.0


<h4 align="center"> Tabela 1: Tabela completa</h4>
    
| Nota | Frequencia  | Frequencia relativa | 
| :---: | :---: | :---: | 
| 0 a 5 | 1 | 6.67 | 
| 5 a 7 | 4 | 26.67 | 
| 7 a 9 | 5 | 33.33 | 
| 9 a 10 | 5 | 33.33 | 
| Total| 15 | 1,0 |

2) Especifique os tipos das variaveis:

    a) peso de um lutador de sumo; 
    b) marcas de motos; 
    c) resistencia de materiais; 
    d) quantidade de chuvas em Pocos de Caldas;
    e) nacionalidade;
    f) numero de versos em um poema;
    g) numero de acoes da Petrobras vendidas no dia;
    h) comprimento de terrenos;
    i) numero de mortes por ataques de tubarao no ano;
    j) resistencia a tracao de metais em kgf/mm^2;
    k) temperatura de ponto de ebulicao da agua, em graus Celsius;
    l) numero de itens nao conformes de uma linha de producao;
    m) tensao em Newtons de um fio de algodao esticado;
    n) numero de livros vendidos no mes.

In [5]:
# Deixe seus calculos aqui:
#a) Numérica contínua
#b) Categorizada nominal
#c) Numérica contínua
#d) Numérica contínua
#e) Categorizada nominal
#f) numérica discreta
#g) numérica discreta
#h) numérica contícua
#i) numérica discreta
#j) numérica contínua
#k) numérica discreta
#l) numérica discreta
#m) numérica discreta
#n) numérica discreta

3) Complete a tabela abaixo.

<h4 align="center"> Tabela 1: Tabela para completar</h4>
    
| Nota | Frequencia  | Frequencia relativa | 
| :---: | :---: | :---: | 
| 9 a 10 | - | 0,08 | 
| 8 a 8,9 | 36 | - | 
| 6,5 a 7,9 | 90 | - | 
| 5 a 6,4 | 30 | - | 
| abaixo de 5 | 28 | - | 
| Total| 200 | 1,0 |


<h4 align="center"> Tabela 2: Tabela completa</h4>
    
| Nota | Frequencia  | Frequencia relativa | 
| :---: | :---: | :---: | 
| 9 a 10 | 16 | 0,08 | 
| 8 a 8,9 | 36 | 0,18 | 
| 6,5 a 7,9 | 90 | 0,45 | 
| 5 a 6,4 | 30 | 0,15 | 
| abaixo de 5 | 28 | 0,14 | 
| Total| 200 | 1,0 |

In [6]:
# Deixe seus calculos aqui:
#total mensos as frequencias
total = 200 
freq = total - 28 - 30 - 90 -36
print(freq)

frequencia_relativa_linha_2 = 36/total
frequencia_relativa_linha_3 = 90/total
frequencia_relativa_linha_4 = 30/total
frequencia_relativa_linha_5 = 28/total

print(f"{(frequencia_relativa_linha_2):0.2f}")
print(f"{(frequencia_relativa_linha_3):0.2f}")
print(f"{(frequencia_relativa_linha_4):0.2f}")
print(f"{(frequencia_relativa_linha_5):0.2f}")

16
0.18
0.45
0.15
0.14


4) Em uma votacao com 3 candidatos, os votos foram:

Bruschi, Allan, Allan, Conte, Allan, Bruschi, Bruschi, Bruschi, Allan, Allan, Conte, Allan, Bruschi, Bruschi, Allan, Conte, Bruschi, Bruschi, Bruschi, Allan

a) Determine o numero de votos de cada pessoa;<br>
b) Calcule as porcentagens;<br>
c) Verifique se algum dos candidatos teve maioria absoluta (50% + 1).

In [7]:
# Deixe seus calculos aqui:
votos = ['Bruschi', 'Allan', 'Allan', 'Conte', 'Allan', 'Bruschi', 'Bruschi', 
         'Bruschi', 'Allan', 'Allan', 'Conte', 'Allan', 'Bruschi', 'Bruschi', 
         'Allan', 'Conte', 'Bruschi', 'Bruschi', 'Bruschi', 'Allan']
print(votos)

#a) Determine o numero de votos de cada pessoa;
frequencia_conte   = sum(1 for item in votos if item == "Conte")
frequencia_bruschi = sum(1 for item in votos if item == "Bruschi")
frequencia_allan   = sum(1 for item in votos if item == "Allan")

print("\nVotos para Conte:",frequencia_conte)
print("Votos para Bruschi:",frequencia_bruschi)
print("Votos para Allan:",frequencia_allan)

#b) Calcule as porcentagens;
total_votos = frequencia_conte + frequencia_bruschi + frequencia_allan;

frequencia_relativa_conte   = frequencia_conte / total_votos
frequencia_relativa_bruschi = frequencia_bruschi / total_votos
frequencia_relativa_allan   = frequencia_allan / total_votos
print("\nPorcentagem Conte:",frequencia_relativa_conte)
print("Porcentagem Bruschi:",frequencia_relativa_bruschi)
print("Porcentagem Allan:",frequencia_relativa_allan)

#c) Verifique se algum dos candidatos teve maioria absoluta (50% + 1).


['Bruschi', 'Allan', 'Allan', 'Conte', 'Allan', 'Bruschi', 'Bruschi', 'Bruschi', 'Allan', 'Allan', 'Conte', 'Allan', 'Bruschi', 'Bruschi', 'Allan', 'Conte', 'Bruschi', 'Bruschi', 'Bruschi', 'Allan']

Votos para Conte: 3
Votos para Bruschi: 9
Votos para Allan: 8

Porcentagem Conte: 0.15
Porcentagem Bruschi: 0.45
Porcentagem Allan: 0.4


In [8]:
import pandas as pd

#a) Determine o numero de votos de cada pessoa;
print("Numero de votos de cada pessoa")
print(pd.Series(votos).value_counts());

#b) Calcule as porcentagens;
print("\nTotal em  porcentagens")
#print(((pd.Series(votos) / len(votos))).to_string())

#c) Verifique se algum dos candidatos teve maioria absoluta (50% + 1).


Numero de votos de cada pessoa
Bruschi    9
Allan      8
Conte      3
dtype: int64

Total em  porcentagens


5) 40 doadores de sangue doaram para o hospital uma bolsa cada, conforme a tabela de tipos sanguineos:

B, A, O, A, A, A, B, O, B, A, A, AB, O, O, A, O, O, A, A, A, B, A, A, A, O, O, O, A, O, A, O, O, A, O, AB, O, O, A, AB, B, B.

Coloque os dados em uma tabela de distribuicao de frequencias.

In [9]:
# Deixe seus calculos aqui:

dados = ['B', 'A', 'O', 'A', 'A', 'A', 'B', 'O', 'B', 'A', 'A', 'AB', 'O', 'O', 'A', 'O', 
         'O', 'A', 'A', 'A', 'B', 'A', 'A', 'A', 'O', 'O', 'O', 'A', 'O', 'A', 'O', 'O', 
         'A', 'O', 'AB', 'O', 'O', 'A', 'AB', 'B', 'B']

#frequencia = soma de elementos de mesmo valor
frequencia_a  = sum(1 for item in dados if item == 'A')
frequencia_b  = sum(1 for item in dados if item == 'B')
frequencia_ab = sum(1 for item in dados if item == 'AB')
frequencia_o  = sum(1 for item in dados if item == 'O')

print("Frequencia A:" ,frequencia_a)
print("Frequencia B:" ,frequencia_b)
print("Frequencia AB:",frequencia_ab)
print("Frequencia O:" ,frequencia_o)

#Calcula frequencia relativa
print("_"*30)
frequencia_relativa_a = frequencia_a / len(dados)
frequencia_relativa_b = frequencia_b / len(dados)
frequencia_relativa_ab = frequencia_ab / len(dados)
frequencia_relativa_o = frequencia_o / len(dados)
print(f"Frequencia Relativa A: {(frequencia_relativa_a ):0.2f}")
print(f"Frequencia Relativa B: {(frequencia_relativa_b ):0.2f}")
print(f"Frequencia Relativa AB: {(frequencia_relativa_ab ):0.2f}")
print(f"Frequencia Relativa O: {(frequencia_relativa_o ):0.2f}")

print("_"*30)
total = frequencia_relativa_a + frequencia_relativa_b + frequencia_relativa_ab + frequencia_relativa_o
print(total)


Frequencia A: 17
Frequencia B: 6
Frequencia AB: 3
Frequencia O: 15
______________________________
Frequencia Relativa A: 0.41
Frequencia Relativa B: 0.15
Frequencia Relativa AB: 0.07
Frequencia Relativa O: 0.37
______________________________
0.9999999999999999


<h4 align="center"> Tabela 2: Tabela completa</h4>
    
| TipoSanquineo | Frequencia  | Frequencia relativa | 
| :---: | :---: | :---: | 
| A | 17 | 0,41 | 
| B | 6 | 0,15 | 
| AB | 3 | 0,07 | 
| O | 15 | 0,37 | 
| Total| 41 | 1,0 |

6) 80 alunos do curso de pos graducao tiveram o seguinte desempenho: 25% receberam nota A, 70% receberam nota B, e 5% tiveram nota C. Qual a frequencia dos alunos com nota A?

In [10]:
# Deixe seus calculos aqui:
total = 80
frequencia_relativa_a = 25/100
frequencia_relativa_b = 70/100
frequencia_relativa_c = 5/100

frequencia_a =  total * frequencia_relativa_a
frequencia_b =  total * frequencia_relativa_b
frequencia_c =  total * frequencia_relativa_c

print(frequencia_a)
print(frequencia_b)
print(frequencia_c)

20.0
56.0
4.0


<h4 align="center"> Tabela 2: Tabela completa</h4>
    
| Desempelho | Frequencia  | Frequencia relativa | 
| :---: | :---: | :---: | 
| A | 20 | 0,25 | 
| B | 56 | 0,70 | 
| C | 4 | 0,05 | 
| Total| 80 | 1,0 |

7) Para que um determinado trecho em uma rota aeronautica seja lucrativo, um certo numero de assentos ocupados 
seria necessario. As classes de estudo para assentos desocupados foram: <br>
    0 a 4; 5 a 9; 10 a 14, 15 a 19, 20 ou mais.<br> 
    Com base nessas informacoes, seria possivel obter a quantidade de voos em que:
<br>
a) havia menos que 10 assentos desocupados?<br>
b) mais de 14 acentos desocupados?<br>
c) pelo menos 5 assentos desocupados?<br>
d) exatamente 9 assentos desocupados?

In [11]:
# Deixe suas respostas aqui:

#a)Sim, somar as frequencias das duas classes anteriores
#b)Sim, somar as frequencias da 2 classes posteriores
#c)Sim, com exceçao da primieria classe
#d)Não, pois temos os valores em intervalos

8) Em uma sala com fanaticos por motos, foram obtidos os seguintes dados para preferencias: 19 pessoas 
    gostariam de ter uma moto esportiva com carenagem, 13 desejam uma moto custom, 
    3 querem esportivas naked, 2 querem uma scooter, 2 preferem motos de trilhas, e 1 deseja um quadriciclo. 
    Monte uma tabela com distribuicao de frequencias.

In [12]:
total = 40
frequencia_relativa_esportiva = (19/total)
frequencia_relativa_custom = (13/total)
frequencia_relativa_naked = (3/total)
frequencia_relativa_scooter = (2/total)
frequencia_relativa_trilha = (2/total)
frequencia_relativa_quadricilho = (total)

print(f'frequencia relativa esportiva: {(frequencia_relativa_esportiva):0.2f}')
print(f'frequencia relativa esportiva: {(frequencia_relativa_custom):0.2f}')
print(f'frequencia relativa esportiva: {(frequencia_relativa_naked):0.2f}')
print(f'frequencia relativa esportiva: {(frequencia_relativa_scooter):0.2f}')
print(f'frequencia relativa esportiva: {(frequencia_relativa_trilha):0.2f}')
print(f'frequencia relativa esportiva: {(frequencia_relativa_quadricilho):0.2f}')

frequencia relativa esportiva: 0.47
frequencia relativa esportiva: 0.33
frequencia relativa esportiva: 0.07
frequencia relativa esportiva: 0.05
frequencia relativa esportiva: 0.05
frequencia relativa esportiva: 40.00


 Deixe seus calculos aqui:

<h4 align="center"> Tabela 2: Tabela completa</h4>
    
| Preferencia | Frequencia  | Frequencia relativa | 
| :---: | :---: | :---: | 
| Esportiva |19 | 0,02 | 
| Custom | 13 | 0,03 | 
| Naked | 3 | 0,13 | 
| Scooter | 2 | 0,20 | 
| Trilha | 2 | 0,20 | 
| Quadriciclo | 1 | 0,40 | 
| Total| 40 | 1,0 |



9) Uma competicao com 100 atletas, ha 7 com 1 medalha, 12 com 2 medalhas, 23 com 3 medalhas, 43 com 4 medalhas,
8 com 5 medalhas, 6 com 6 medalhas e 3 com 7 medalhas. Apresente os dados em uma tabela.

In [13]:
# Deixe seus calculos aqui:


10) O setor de qualidade de uma fabrica de chocolates quer verificar se o peso de 65 gramas de um produto esta
de acordo com o informado na embalagem. Em uma busca aleatoria, obteve-se os seguintes dados:
    
<h4 align="center"> Tabela 3: Peso dos produtos, em gramas</h4>
  
| | | | |    
| :---: | :---: | :---: | :---: |
| 63,6 | 64,0 | 64,8 | 65,2 | 
| 63,7 | 64,0 | 64,8 | 65,3 |
| 63,7 | 64,2 | 64,9 | 65,3 | 
| 63,7 | 64,6 | 65,1 | 65,4 | 
| 63,7 | 64,7 | 65,2 | 65,4 | 


Construa uma tabela de distribuicao de frequencias.
    

In [14]:
import pandas as pd

In [15]:
# Deixe seus calculos aqui:
pesos = ['63.6','64.0','64.8','65.2',
         '63.7','64.0','64.8','65.3',
         '63.7','64.2','64.9','65.3',
         '63.7','64.6','65.1','65.4',
         '63.7','64.7','65.2','65.4']

frequencia = pd.Series(pesos).value_counts()
print(f'Frequencia: \n{frequencia}')

frequencia_relativa=[(float(p)/(len(pesos))) for p in frequencia]  
print(f'\nFrequenciaRelativa: \n{frequencia_relativa}')

tabela = []
for p, f, fr in zip(pesos, frequencia, frequencia_relativa):
    tabela.append([p, f, fr])
print(f'\nTabela: \n{tabela}')

Frequencia: 
63.7    4
65.2    2
65.3    2
64.0    2
65.4    2
64.8    2
64.6    1
65.1    1
64.7    1
64.9    1
63.6    1
64.2    1
dtype: int64

FrequenciaRelativa: 
[0.2, 0.1, 0.1, 0.1, 0.1, 0.1, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05]

Tabela: 
[['63.6', 4, 0.2], ['64.0', 2, 0.1], ['64.8', 2, 0.1], ['65.2', 2, 0.1], ['63.7', 2, 0.1], ['64.0', 2, 0.1], ['64.8', 1, 0.05], ['65.3', 1, 0.05], ['63.7', 1, 0.05], ['64.2', 1, 0.05], ['64.9', 1, 0.05], ['65.3', 1, 0.05]]


| Peso | Frequencia  | Frequencia relativa | 
| :---: | :---: | :---: | 
| 63.6 | 4 | 0.2 | 
| 64.0 | 2 | 0.2 | 
| 64.8 | 2 | 0.1 | 
| 65.2 | 2 | 0.1 | 
| 63.7 | 2 | 0.1 | 
| 64.0 | 2 | 0.1 | 
| 64.8 | 1 | 0.05 | 
| 65.3 | 1 | 0.05 | 
| 63.7 | 1 | 0.05 | 
| 64.2 | 1 | 0.05 | 
| 64.9 | 1 | 0.05 | 
| 65.3 | 1 | 0.05 | 
| Total| 20| 1 |

11) Determine o tipo da variavel:


a) dados de contagem (discretos ou continuos);<br>
b) numero de vezes que um evento acontece (frequencia, frequencia relativa, total)<br>
c) cores de cabelos (ordinais, nominais, continuos, discretos).

In [16]:
# Deixe suas respostas aqui:
#a)Discreto
#b)Frequenica
#c)Nominal

12) Verdadeiro ou falso?
    
    a) frequencia relativa e igual a frequencia dividida pelo total.
    b) estatistica e a arte de apresentar graficos.

In [17]:
# Deixe suas respostas aqui:
#a)Verdadeiro
#b)Falso

13) Para divulgacao de aprovados ou reprovados em uma disciplina, organize na forma de tabela de frequencias
as notas dos 15 alunos do curso da seguinte maneira: D para notas de 0 a 5 (reprovado); C para notas de 5 a 7;
B para notas de 7 a 9; e A para notas acima de 9.

dados: 7, 8, 6, 5, 6, 4, 7, 10, 7, 8, 8, 3, 8, 9, 10

In [18]:
# Deixe seus calculos aqui:
notas = [7, 8, 6, 5, 6, 4, 7, 10, 7, 8, 8, 3, 8, 9, 10]

total = 15
D=[i for i in notas if (i>=0 and i<5)] 
C=[i for i in notas if (i>=5 and i<7)] 
B=[i for i in notas if (i>=7 and i<9)]
A=[i for i in notas if (i>=9)]

frequenciaD = len(D)
frequenciaB = len(C)
frequenciaC = len(B)
frequenciaA = len(A)

print(f'Frequencia D:{frequenciaD}')
print(f'Frequencia B:{frequenciaC}')
print(f'Frequencia C:{frequenciaB}')
print(f'Frequencia A:{frequenciaA}')

frequencia_relativa_D = frequenciaD/total
frequencia_relativa_C = frequenciaC/total
frequencia_relativa_B = frequenciaB/total
frequencia_relativa_A = frequenciaA/total

print(f'Frequencia Relativa D:{frequencia_relativa_D}')
print(f'Frequencia Relativa B:{frequencia_relativa_C}')
print(f'Frequencia Relativa C:{frequencia_relativa_B}')
print(f'Frequencia Relativa A:{frequencia_relativa_A}')

frequencia_D =  total * frequencia_relativa_D
frequencia_C =  total * frequencia_relativa_C
frequencia_B =  total * frequencia_relativa_B
frequencia_A =  total * frequencia_relativa_A



Frequencia D:2
Frequencia B:7
Frequencia C:3
Frequencia A:3
Frequencia Relativa D:0.13333333333333333
Frequencia Relativa B:0.4666666666666667
Frequencia Relativa C:0.2
Frequencia Relativa A:0.2


| Intervalo | Frequencia  | Frequencia relativa | 
| :---: | :---: | :---: | 
| D | 2 | 0,13 | 
| B | 7 | 0,46 | 
| C | 3 | 0,02 | 
| A | 3 | 0,02 | 
| Total| 15 | 1,0 |

14) O setor de qualidade de uma fabrica obteve a seguinte tabela com dados de embalagens com defeitos.

<h4 align="center"> Tabela 4: Estado de embalagens</h4>
  
| | | | | | 
| :---: | :---: | :---: | :---: | :---: |
| amassada | sem defeito  | sem defeito | amassada | sem defeito  |
| sem defeito  | cortada | amassada | sem defeito | sem defeito  |
| nao fechada | sem defeito  | sem defeito  | sem defeito  | sem defeito  |
| sem defeito | sem defeito | sem defeito | sem defeito  | sem defeito | 
| sem defeito  | sem defeito | sem defeito  | amassada | sem defeito | 

a) Construa uma tabela de distribuicao de frequencias;<br>
b) Explique se ha necessidade de melhorias no processo de embalagem.


In [19]:
# Deixe seus calculos aqui:
esdado_das_embalagens = ['amassada','sem defeito','sem defeito','amassada','sem defeito',
                         'sem defeito','cortada','amassada','sem defeito','sem defeito',
                         'nao fechada','sem defeito','sem defeito','sem defeito','sem defeito',
                         'sem defeito','sem defeito','sem defeito','sem defeito','sem defeito',
                         'sem defeito','sem defeito','sem defeito','amassada','sem defeito']
total=len(esdado_das_embalagens)
frequencia_amassada    = sum(1 for item in esdado_das_embalagens if item == 'amassada')
frequencia_sem_defeito = sum(1 for item in esdado_das_embalagens if item == 'sem defeito')
frequencia_cortada     = sum(1 for item in esdado_das_embalagens if item == 'cortada')
frequencia_nao_fechada = sum(1 for item in esdado_das_embalagens if item == 'nao fechada')

print(f'Frequencia amassada\t:{frequencia_amassada}')
print(f'Frequencia sem defeito\t:{frequencia_sem_defeito}')
print(f'Frequencia cortada\t:{frequencia_cortada}')
print(f'Frequencia nao fechada\t:{frequencia_nao_fechada}')
print(total)

frequenciaR_amassada    = frequencia_amassada/total
frequenciaR_sem_defeito = frequencia_sem_defeito/total
frequenciaR_cortada     = frequencia_cortada/total
frequenciaR_nao_fechada = frequencia_nao_fechada/total

print("_"*30)
print(f'Frequencia relativa amassada\t:{frequenciaR_amassada}')
print(f'Frequencia relativa sem defeito\t:{frequenciaR_sem_defeito}')
print(f'Frequencia relativa cortada\t:{frequenciaR_cortada}')
print(f'Frequencia relativa nao fechada\t:{frequenciaR_nao_fechada}')

print("_"*30)
total = frequenciaR_amassada + frequenciaR_sem_defeito + frequenciaR_cortada + frequenciaR_nao_fechada
print(total)

frequencia_D =  total * frequencia_relativa_D
frequencia_C =  total * frequencia_relativa_C
frequencia_B =  total * frequencia_relativa_B
frequencia_A =  total * frequencia_relativa_A



Frequencia amassada	:4
Frequencia sem defeito	:19
Frequencia cortada	:1
Frequencia nao fechada	:1
25
______________________________
Frequencia relativa amassada	:0.16
Frequencia relativa sem defeito	:0.76
Frequencia relativa cortada	:0.04
Frequencia relativa nao fechada	:0.04
______________________________
1.0


| Peso | Frequencia  | Frequencia relativa | 
| :---: | :---: | :---: | 
| amassada | 4 | 0,16 | 
| sem defeito | 19 | 0,76 | 
| cortada | 1 | 0,04 | 
| nao fechada | 1 | 0,04 | 
| Total| 25 | 1,0 |

15) O setor de qualidade da fabrica anterior resolveu verificar a eficiencia de duas turmas de trabalho.
A turma A teve uma producao de 563 itens conformes, e 151 itens nao conformes. A turma B produziu 307 itens conformes e 85 nao conformes. Organize os dados em uma tabela, mostrando a porcentagem de conformes e nao conformes.

In [20]:
# Deixe seus calculos aqui:
turma_a_conforme = 563
turma_a_nao_conforme = 151
turma_b_conforme = 307
turma_b_nao_conforme = 85

turma_a_total = turma_a_conforme + turma_a_nao_conforme
turma_b_total = turma_b_conforme + turma_b_nao_conforme 
print(turma_a_total)
print(turma_b_total)

turma_a_porcentagem_conforme     = turma_a_conforme/turma_a_total*100
turma_a_porcentagem_nao_conforme = turma_a_nao_conforme/turma_a_total*100

turma_b_porcentagem_conforme     = turma_b_conforme/turma_b_total*100
turma_b_porcentagem_nao_conforme = turma_b_nao_conforme/turma_b_total*100 


print(f'Turma A % conforme: {turma_a_porcentagem_conforme:0.2f}')
print(f'Turma A % não conforme: {turma_a_porcentagem_nao_conforme:0.2f}')
      
print(f'Turma B % conforme: {turma_b_porcentagem_conforme:0.2f}')
print(f'Turma B % não conforme: {turma_b_porcentagem_nao_conforme:0.2f}')



714
392
Turma A % conforme: 78.85
Turma A % não conforme: 21.15
Turma B % conforme: 78.32
Turma B % não conforme: 21.68


| Produçao | TurmaA  | TurmaB | 
| :---: | :---: | :---: | 
| Conforme | 563 | 307 | 
| Nao Conforme | 151 |85|
| Total| 714 | 392 | 

| Produçao | TurmaA  | TurmaB | 
| :---: | :---: | :---: | 
| Conforme |  78.85% | 78.32%| 
| Nao Conforme | 21.15% |21.68% |

16) Analisando os dados de acidentes de trabalho no periodo de 3 anos da fabrica citada anteriormente, 
obteve-se a seguinte tabela:

<h4 align="center"> Tabela 5: Acidentes por departamento e dias da semana</h4>
  
| DEPARTAMENTO | segunda-feira | terca-feira | quarta-feira | quinta-feira | sexta-feira | 
| :---: | :---: | :---: | :---: | :---: | :---: |
| A | 12 | 21 | 23 | 29  | 35 |
| B  | 14 | 10 | 11 | 10 | 15 |
| C | 12  | 12  | 13 | 11 | 11 |
| D | 24 | 22 | 23 | 21 | 22 |

a) Em que dia da semana ocorreu mais acidentes?<br>
b) Em que departamento?

In [21]:
import operator

In [22]:
# Deixe seus calculos aqui:[
acidentes = [[12,21,23,29,35],
             [14,10,11,10,15],
             [12,12,13,11,11],
             [24,22,23,21,22]]

departamentos_acidentes = [{'departamento':'a','acidentes':sum(acidentes[0])},
                           {'departamento':'b','acidentes':sum(acidentes[1])},
                           {'departamento':'c','acidentes':sum(acidentes[2])},
                           {'departamento':'d','acidentes':sum(acidentes[3])}]
print(departamentos_acidentes)

min_price = min(item['acidentes'] for item in departamentos_acidentes)
print(f'Menor numero de acientes: {min_price}')

max_price = max(item['acidentes'] for item in departamentos_acidentes)
print(f'Maior numero de acientes: {max_price}')

[{'departamento': 'a', 'acidentes': 120}, {'departamento': 'b', 'acidentes': 60}, {'departamento': 'c', 'acidentes': 59}, {'departamento': 'd', 'acidentes': 112}]
Menor numero de acientes: 59
Maior numero de acientes: 120


In [6]:
from functools import reduce
acidentes_dia_semana = [[12,14,12,24],
                        [21,10,12,22],
                        [23,11,13,23],
                        [29,10,11,21],
                        [35,15,11,22]]

dia_semana_acidentes = [{'DiaSemana':'segunda-feira','acidentes':sum(acidentes_dia_semana[0])},
                        {'DiaSemana':'terca-feira','acidentes':sum(acidentes_dia_semana[1])},
                        {'DiaSemana':'quarta-feira','acidentes':sum(acidentes_dia_semana[2])},
                        {'DiaSemana':'quinta-feira','acidentes':sum(acidentes_dia_semana[3])},
                        {'DiaSemana':'sexta-feira','acidentes':sum(acidentes_dia_semana[4])}]
print(dia_semana_acidentes)

dis_semana_min_acidente = min(item['acidentes'] for item in dia_semana_acidentes)
print(f'Menor numero de acientes por dia da semana: {dis_semana_min_acidente}')

dis_semana_max_acidente = max(item['acidentes'] for item in dia_semana_acidentes)
print(f'Maior numero de acientes por dia da semana: {dis_semana_max_acidente}')

[{'DiaSemana': 'segunda-feira', 'acidentes': 62}, {'DiaSemana': 'terca-feira', 'acidentes': 65}, {'DiaSemana': 'quarta-feira', 'acidentes': 70}, {'DiaSemana': 'quinta-feira', 'acidentes': 71}, {'DiaSemana': 'sexta-feira', 'acidentes': 83}]
Menor numero de acientes por dia da semana: 62
Maior numero de acientes por dia da semana: 83


17) A distribuicao dos salarios de 200 funcionarios, em multiplos de RS 1000,00, segue conforme a tabela abaixo:

<h4 align="center"> Tabela 6: Distribuicao de salarios de 200 funcionarios</h4>
  
| Classe | Porcentagem | 
| :---: | :---: | 
| 2 a 4 | 20 | 
| 4 a 5 | 25 | 
| 5 a 8 | 45  | 
| 9 a 10 | 10 | 
| Total | 100 | 

O numero de funcionarios que possuem salarios maiores ou iguais a RS 4000,00 e inferiores a RS 8000,00 e:

a) 60<br>
b) 80<br>
c) 90<br>
d) 140<br>
e) 160

In [23]:
# Deixe seus calculos aqui:
total_funcionarios = 200
porcentagem_no_intervalo = 70 / 100 #(25 + 45) Classe 4 a 5 e Classe 4 a 8  
qtd_funcionarios_no_intervalo = total_funcionarios*porcentagem_no_intervalo

print(f'Numero de funcionario com salario entre 4 e 8 mil: {qtd_funcionarios_no_intervalo}')



Numero de funcionario com salario entre 4 e 8 mil: 140.0


18) Uma cidade de 10 mil habitantes gera aproximadamente 0,76 kg de lixo por pessoa em um dia, mas somente 82% e atendido
pelo aterro sanitario da cidade. A quantidade de lixo que realmente acaba indo para o aterro, em kg/dia, e: <br>

a) 1368; <br> 
b) 1558; <br>
c) 6232; <br>
d) 7600; <br>
e) 24928.

In [24]:
# Deixe seus calculos aqui:
total_habitantes = 10000
lixo_pessoa_dia = 0.76
atendimento_aterro = 82/100

qtd_diaria_aterro = (total_habitantes*lixo_pessoa_dia)*atendimento_aterro
print(f'Quantidade diaria que realmente vai para o aterro: {qtd_diaria_aterro}')

Quantidade diaria que realmente vai para o aterro: 6232.0
